akka-http2-support/src/main/scala/akka/http/impl/engine/http2/hpack/HeaderDecompression.scala

```diff
@@ -55,12 +54,24 @@ private[http2] object HeaderDecompression extends GraphStage[FlowShape[FrameEven
       } catch {
         case ex: IOException ⇒
           // this is signalled by the decoder when it failed, we want to react to this by rendering a GOAWAY frame
-          fail(eventsOut, new Http2Compliance.HeaderDecompressionFailed("Decompression failed."))
+          fail(eventsOut, new Http2Compliance.HeaderDecompressionFailed("Header decompression failed."))
       }
     }
 
+    // TODO optimise this
+    def handleSettings: PartialFunction[FrameEvent, Unit] = {
+      case frame @ SettingsFrame(settings) ⇒
+        settings.collect {
+          case Setting(SettingIdentifier.SETTINGS_HEADER_TABLE_SIZE, size) ⇒
```

Comment:
That's again the wrong way round. We need to call `decoder.setMaxHeaderTableSize` when we receive the ack for this setting we sent ourselves (which we don't do currently).